sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -434,6 +434,17 @@ case class RangeExec(range: org.apache.spark.sql.catalyst.plans.logical.Range)
     val input = ctx.freshName("input")
     // Right now, Range is only used when there is one upstream.
     ctx.addMutableState("scala.collection.Iterator", input, s"$input = inputs[0];")
+
+    val localIdx = ctx.freshName("localIdx")
+    val localEnd = ctx.freshName("localEnd")
+    val range = ctx.freshName("range")
+    // we need to place consume() before calling isShouldStopRequired
+    val body = consume(ctx, Seq(ev))
+    val shouldStop = if (isShouldStopRequired) {
```

Comment:
`isShouldStopRequired` complicates the logic. Is it necessary? How much improvement it brings?